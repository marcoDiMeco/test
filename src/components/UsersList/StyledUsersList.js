import styled from "styled-components";

const StyledUsersList = styled.div`
  display: flex;
  align-items: start;
  justify-content: center;
  flex-wrap: wrap;
`;

const StyledUserContainer = styled.div`
  width: 100%;
  max-width: 200px;
  text-align: center;
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 20px;
  flex-wrap: wrap;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-left: 40px;
  padding-right: 40px;
  background: white;
`;

const StyledAvatar = styled.div`
  width: 100px;
  height: 100px;
  border-radius: 50%;
  border: 2px solid #f4f4f4;
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
  margin-bottom: 10px;
`;

const StyledName = styled.div`
  font-size: 15px;
  width: 100%;
`;

const StyledSurname = styled.div`
  font-size: 15px;
  font-weight: 700;
  width: 100%;
`;

const StyledLoadingContainer = styled.div`
  width: 100%;
  padding-top: 10px;
  padding-bottom: 100px;
  display: flex;
  justify-content: center;
`;

const StyledError = styled.div`
  width: 100%;
  padding-top: 100px;
  font-size: 18px;
  line-height: 25px;
  font-weight: 700;
  color: #e70505;;
  display: flex;
  justify-content: center;
  text-align: center;
`;

export {
  StyledUserContainer,
  StyledAvatar,
  StyledName,
  StyledUsersList,
  StyledSurname,
  StyledLoadingContainer,
  StyledError,
};
