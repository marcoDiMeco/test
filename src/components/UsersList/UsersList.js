import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { Loading } from "../";
import { AMOUNT_VALUE } from "../../constants";
import {
  StyledUsersList,
  StyledUserContainer,
  StyledAvatar,
  StyledName,
  StyledSurname,
  StyledLoadingContainer,
  StyledError,
} from "./StyledUsersList";

const UsersList = ({
  data,
  isLoading,
  getMoreUsers,
  list,
  isPaginationLoading,
  error,
}) => {
  const [isBottom, setIsBottom] = useState(false);

  useEffect(() => {
    if (list === "second") {
      window.addEventListener("scroll", handleScroll);
      return () => window.removeEventListener("scroll", handleScroll);
    }
  }, [list]);

  useEffect(() => {
    if (isBottom && data.length < parseInt(AMOUNT_VALUE.first)) {
      getMoreUsers(AMOUNT_VALUE.second);
    }
  }, [isBottom, getMoreUsers, data]);

  const handleScroll = () => {
    const scrollTop =
      (document.documentElement && document.documentElement.scrollTop) ||
      document.body.scrollTop;
    const scrollHeight =
      (document.documentElement && document.documentElement.scrollHeight) ||
      document.body.scrollHeight;
    if (scrollTop + window.innerHeight >= scrollHeight) {
      setIsBottom(true);
    }
  };

  return (
    <StyledUsersList>
      {error && (
        <StyledError>
          An error occurred.
          <br />
          Please refresh page.
        </StyledError>
      )}
      {isLoading ? (
        <Loading isBottom={false} />
      ) : (
        data.map((item, index) => {
          return (
            <StyledUserContainer key={index}>
              <StyledAvatar
                style={{ backgroundImage: `url(${item.picture.large})` }}
              />
              <StyledName>{item.name.first}</StyledName>
              <StyledSurname>{item.name.last}</StyledSurname>
            </StyledUserContainer>
          );
        })
      )}
      {isPaginationLoading && (
        <StyledLoadingContainer>
          <Loading isBottom={true} />
        </StyledLoadingContainer>
      )}
    </StyledUsersList>
  );
};

UsersList.propTypes = {
  data: PropTypes.array,
  isLoading: PropTypes.bool,
  getMoreUsers: PropTypes.func,
  list: PropTypes.string,
  isPaginationLoading: PropTypes.bool,
  error: PropTypes.bool,
};

UsersList.defaultProps = {
  data: [],
  isLoading: false,
  list: "first",
  isPaginationLoading: false,
  error: false,
};

export default UsersList;
