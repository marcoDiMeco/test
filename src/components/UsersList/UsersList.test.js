import React from "react";
import renderer from "react-test-renderer";
import UsersList from "./";

describe("UsersList component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<UsersList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
