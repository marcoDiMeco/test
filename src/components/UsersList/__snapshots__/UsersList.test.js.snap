// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList component renders correctly 1`] = `
<div
  className="sc-AxheI cxMVRd"
/>
`;
